File: app/src/main/java/com/story_tail/privypics/ui/login/viewmodel/LogInViewModel.java
package com.story_tail.privypics.ui.login.viewmodel;

import androidx.lifecycle.ViewModel;

public class LogInViewModel extends ViewModel {
    private String email;
    private String password;

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void onSignInClick() {

    }
}

File: app/src/main/java/com/story_tail/privypics/ui/login/viewmodel/LogInViewModel.kt
package com.story_tail.privypics.ui.login.viewmodel

import androidx.lifecycle.ViewModel

class LogInViewModel(private val listener: LogInListener) : ViewModel() {
    var email: String? = null
    var password: String? = null

    fun onSignInClick() {
        listener.onLogInClick(email, password)
    }
}
